openshift/tests-extension/test/olmv1.go

```diff
@@ -2,15 +2,99 @@ package test
 
 import (
 	"context"
+	"fmt"
 
-	//nolint:staticcheck // ST1001: dot-imports are acceptable here for readability in test code
+	//nolint:staticcheck // ST1001: dot-imports for readability
 	. "github.com/onsi/ginkgo/v2"
-	//nolint:staticcheck // ST1001: dot-imports are acceptable here for readability in test code
+	//nolint:staticcheck // ST1001: dot-imports for readability
 	. "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	apiextclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github/operator-framework-operator-controller/openshift/tests-extension/test/env"
+	"github/operator-framework-operator-controller/openshift/tests-extension/test/extlogs"
+)
+
+const (
+	olmv1GroupName = "olm.operatorframework.io"
 )
 
 var _ = Describe("[sig-olmv1] OLMv1", func() {
 	It("should pass a trivial sanity check", func(ctx context.Context) {
 		Expect(len("test")).To(BeNumerically(">", 0))
+
 	})
 })
+
+var _ = Describe("[sig-olmv1][OCPFeatureGate:NewOLM] OLMv1 CRDs", func() {
+	BeforeEach(func() {
+		_ = env.Init()
+	})
+	It("should verify required CRDs are installed and active", func(ctx SpecContext) {
+		checkFeatureCapability(ctx)
+		cfg := env.Get().RestCfg
+		crds := []struct {
+			group   string
+			version []string
+			plural  string
+		}{
+			{olmv1GroupName, []string{"v1"}, "clusterextensions"},
+			{olmv1GroupName, []string{"v1"}, "clustercatalogs"},
+		}
+
+		client, err := apiextclient.NewForConfig(cfg)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, crd := range crds {
+			By(fmt.Sprintf("verifying CRD %s.%s", crd.plural, crd.group))
+			crdObj, err := client.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, fmt.Sprintf("%s.%s",
+				crd.plural, crd.group), metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, v := range crd.version {
+				found := false
+				for _, ver := range crdObj.Spec.Versions {
+					if ver.Name == v {
+						Expect(ver.Served).To(BeTrue(), "version %s not served", v)
+						Expect(ver.Storage).To(BeTrue(), "version %s not used for storage", v)
+						found = true
+						break
+					}
+				}
+				if !found {
+					extlogs.WarnContextf("Expected version not found for CRD %s.%s. Available: %#v",
+						crd.plural,
+						crd.group,
+						crdObj.Spec.Versions)
+				}
+				Expect(found).To(BeTrue(), fmt.Sprintf("Expected version %q in CRD %s.%s", v,
+					crd.plural, crd.group))
+			}
+		}
+	})
+})
```

Comment:
Same test done in : https://github.com/openshift/origin/blob/48d56f2e5bb8bcf6c4560fb077c7b809e18a5c19/test/extended/olm/olmv1.go#L34-L74

(Migration - After we migrate for here and check that all is fine with Sippy we can remove from opc/origin)